src/hotspot/share/gc/g1/g1MonitoringSupport.cpp

```diff
@@ -356,3 +363,23 @@ G1MonitoringScope::~G1MonitoringScope() {
   // Needs to be called after updating pool sizes.
   MemoryService::track_memory_usage();
 }
+
+G1YoungGCMonitoringScope::G1YoungGCMonitoringScope(G1MonitoringSupport* monitoring_support,
+                                                   bool all_memory_pools_affected) :
+  G1MonitoringScope(monitoring_support,
+                    monitoring_support->_incremental_collection_counters,
+                    &monitoring_support->_incremental_memory_manager,
```

Comment:
Preexisting: it's slightly odd that one is a pointer and the other not. I wonder if both can be non-pointer.

Reply:
This can be considered in another CR I think, together with the renaming of these variables, proposed in an earlier comment.